packages/cw20/src/query.rs

```diff
@@ -84,3 +84,60 @@ pub struct AllAllowancesResponse {
 pub struct AllAccountsResponse {
     pub accounts: Vec<HumanAddr>,
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use cosmwasm_schema::schema_for;
+    use serde_json;
+    use valico::json_schema;
+
+    #[test]
+    fn it_should_ser_de() {
+        let response = AllowanceResponse {
+            allowance: Uint128(500000),
+            expires: Expiration::AtTime(1234),
+        };
+
+        let info = AllowanceInfo {
+            allowance_response: response,
+            spender: HumanAddr::from("spender"),
+        };
+        let allowance_info = serde_json::to_value(info.clone()).unwrap();
+        let serialized = serde_json::to_string(&allowance_info).unwrap();
+        let deserialized: AllowanceInfo = serde_json::from_str(&serialized).unwrap();
+        assert!(info.eq(&deserialized));
+    }
+
+    #[test]
+    fn it_should_make_correct_schema() {
```

Comment:
Oh, this is a very interesting idea.

What I would do would be to define the old AllowanceInfo type for tests and generate json schema for the old and new AllowanceInfo (with and without flatten) and you could compare those two schemas (asset equal)